src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -0,0 +1,531 @@
+        credential,
+        load_test_resource,
+        resource_group=resource_group_name,
+        subscription_id=subscription_id,
+    )
+    return admin_data_plane_client(
+        cmd.cli_ctx,
+        subscription=subscription_id,
+        endpoint=endpoint,
+        credential=credential,
+    )
+
+
+def get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name=None):
+    from azext_load.data_plane.client_factory import testrun_data_plane_client
+
+    credential, subscription_id, _ = get_login_credentials(cmd.cli_ctx)
+    endpoint = get_load_test_resource_endpoint(
+        credential,
+        load_test_resource,
+        resource_group=resource_group_name,
+        subscription_id=subscription_id,
+    )
+    return testrun_data_plane_client(
+        cmd.cli_ctx,
+        subscription=subscription_id,
+        endpoint=endpoint,
+        credential=credential,
+    )
+
+
+def get_enum_values(enum):
+    if not isinstance(enum, EnumMeta):
+        raise InvalidArgumentValueError(f"Invalid enum type: {type(enum)}")
+    return [item.value for item in enum]
+
+
+def download_file(url, file_path):
+    logger.debug("Downloading file started")
+    response = None
+    retries = 3
+    ex = None
+    while retries > 0:
+        try:
+            response = requests.get(url, stream=True, allow_redirects=True, timeout=60)
+            break
+        except Exception as e:  # pylint: disable=broad-except
+            ex = e
+            retries -= 1
+            logger.debug(
+                "Exception occurred while downloading file: %s. Retrying the request. Retries remaining: %d",
+                str(ex),
+                retries,
+            )
+    if retries == 0:
+        msg = f"Request for {url} failed after all retries: {str(ex)}"
+        logger.debug(msg)
+        raise FileOperationError(msg)
+
+    if response:
+        with open(file_path, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:  # ignore keep-alive new chunks
+                    f.write(chunk)
+    logger.debug("Downloading file completed")
+
+
+def upload_file_to_test(client, test_id, file_path, file_type=None, wait=False):
+    logger.debug(
+        "Uploading file %s for the test %s with 'wait' %s",
+        file_path,
+        test_id,
+        "enabled" if wait else "disabled",
+    )
+    # pylint: disable-next=protected-access
+    file_path = validators._validate_path(file_path, is_dir=False)
+    with open(file_path, "rb") as file:
+        upload_poller = client.begin_upload_test_file(
+            test_id,
+            file_name=os.path.basename(file.name),
+            file_type=file_type,
+            body=file,
+        )
+        response = (
+            upload_poller.result()
+            if wait
+            else upload_poller.polling_method().resource()
+        )
+        logger.debug(
+            "Upload result for file with --wait%s passed: %s",
```

Comment:
Usually we use `--no-wait` argument instead of `--wait`, because the default behavior for LRO commands is waiting.

Reply:
There are few reasons to move to wait instead of no-wait.

1. no-wait's initial response is not getting returned to output CMD. 
2. Our portal behavior is similar to wait. We won't wait in create command for file upload. 
3. For test run command generally, it can be of any duration (in hours too), so if user wants to wait they should provide wait explicitly otherwise they might get stuck for hours unwillingly.

Reply:
have updated the commands, create and update test have custom no wait and all other commands follow the supports_no_wait flag as you suggested